pkg/init/cmd/service/logging.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	errLoggingNotEnabled = errors.New("logging system not enabled")
+	logWriteSocket       = "/var/run/external-logging.sock"
+	logReadSocket        = "/var/run/memlogdq.sock"
+)
+
+const (
+	logDumpCommand byte = iota
+)
+
+// Log sets up a log destination for a named service.
+type Log interface {
+	Path(name string) string            // Path of the log file (or FIFO)
+	Open(name string) (*os.File, error) // Opens a log file directly
+	DumpAll()                           // Copies all the logs to the console
+}
+
+// GetLog returns the log destination we should use.
+func GetLog(logDir string) Log {
+	// is an external logging system enabled?
+	_, err := os.Stat(logWriteSocket)
+	if !os.IsNotExist(err) {
+		return &remoteLog{
+			fifoDir: "/var/run",
+		}
+	}
+	return &fileLog{
+		dir: logDir,
+	}
+}
+
+type fileLog struct {
+	dir string
+}
+
+// Path returns the name of a log file path for the named service.
+func (f *fileLog) Path(name string) string {
+	// We just need this to exist. If we cannot write to the directory,
+	// we'll discard output instead.
+	file, err := f.Open(name)
+	if err != nil {
+		return "/dev/null"
+	}
+	_ = file.Close()
+	return filepath.Join(f.dir, name+".log")
+}
+
+// Open a log file for the named service.
+func (f *fileLog) Open(name string) (*os.File, error) {
+	return os.Create(filepath.Join(f.dir, name+".log"))
```

Comment:
The result here is `RDWR` rather than `WRONLY` as before. 
I think you can make the return an `io.WriteCloser` here instead of `*os.File` too, which more accurately reflects what the caller wants/needs.